docs/vm-configuration.md

```diff
@@ -264,3 +264,95 @@ the generation of multiple libvirt XML documents, the domain XML and associated
 documents for secrets, network filters and node devices. So there is not a strict
 1-1 mapping between the k8s API resource representation and the libvirt low
 level representation.
+
+Proposed VM Definition
+----------------------
+
+The following definition adheres to main points described above. It provides
+
+1. high level description usable by end users or management applications and
+2. low level details required to construct and launch a VM.
+
+The features section is used to generate the domain section, the other way
+around is out of scope for this proposal. Each feature must represent one or
+more fields in the domain section.
+
+The design and contents of each feature section should be logically split and
+discussed in separate proposal. This is just an example without the sections
+discussed, converted from cluster/vm.yaml. Splitting the features enables us to
+create modular VM Spec generation flows that are easily tunable by experts in
+their domains of expertise.
+
+```yaml
+apiVersion: kubevirt.io/v1alpha2
+kind: VM
+metadata:
+  name: testvm
+spec:
+  features:
```

Comment:
What exactly is the advantage, over just specifying in the spec, what you care about, and let the reset be filled in?

Did you also think about some higher features? There was for instance some discussion, about flavours, ... which could be implemented via admission controllers.

Could you also share, how the same sections look like in the libvirt xml, so that we better see the benefit of this feature section?